pkg/scalers/external_scaler.go

```diff
@@ -112,7 +120,55 @@ func parseExternalScalerMetadata(config *ScalerConfig) (externalScalerMetadata,
 	}
 
 	meta.originalMetadata = make(map[string]string)
+	if val, ok := config.AuthParams["caCert"]; ok {
+		caCertBytes := []byte(val)
+		block, _ := pem.Decode(caCertBytes)
+		if block == nil {
+			return meta, fmt.Errorf("failed to decode PEM-encoded certificate in caCert")
+		}
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			return meta, err
+		}
+		certPool := x509.NewCertPool()
+		certPool.AddCert(cert)
+		meta.caCert = certPool
+	}
 
+	if val, ok := config.AuthParams["tlsClientCert"]; ok {
+		certBytes := []byte(val)
+		block, _ := pem.Decode(certBytes)
+		if block == nil {
+			return meta, fmt.Errorf("failed to decode PEM-encoded certificate in tlsClientCert")
+		}
+		_, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			return meta, fmt.Errorf("failed to parse certificate in tlsClientCert")
+		}
+		meta.tlsClientCert = certBytes
+	}
+
+	if val, ok := config.AuthParams["tlsClientKey"]; ok {
+		keyBytes := []byte(val)
+		block, _ := pem.Decode(keyBytes)
+		if block == nil {
+			return meta, fmt.Errorf("failed to decode PEM-encoded private key in tlsClientKey")
+		}
+		_, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+		if err != nil {
+			return meta, fmt.Errorf("failed to parse private key in tlsClientKey")
+		}
+		meta.tlsClientKey = keyBytes
+	}
+
+	meta.insecureSkipVerify = false
+	if val, ok := config.TriggerMetadata["insecureSkipVerify"]; ok && val != "" {
+		boolVal, err := strconv.ParseBool(val)
+		if err != nil {
+			return meta, fmt.Errorf("failed to parse insecureSkipVerify value. Must be either true or false")
+		}
+		meta.insecureSkipVerify = boolVal
+	}
```

Comment:
I'm not 100% sure about if this is longer needed, as KEDA already supports custom CAs. WDYT @kedacore/keda-core-contributors ?

Reply:
At the beginning, I did not intend to expose this config option as well, because from the official [doc](https://pkg.go.dev/crypto/tls) 
```
This should be used only for
 testing or in combination with VerifyConnection or VerifyPeerCertificate.
```
It is recommended for testing only. Do we foresee any case where user need to set this `insecureSkipVerify` flag ? If not, I think with custom CAs, we should cover most cases. WDYT ?

Reply:
Certanly, other scalers support this with the parameter `unsafeSsl`, maybe we can support it :/ I don't have any strong opinion... Let's wait other folks' opinion (it could need some time because KubeCon is next week)

Reply:
I think we can add this parameter, could be really beneficial for testing/development of a scaler. The external scaler is a special case.

Reply:
Yes. I have added `unsafeSsl` config